pkg/gadget-service/api/api.proto

```diff
@@ -248,7 +258,49 @@ service BuiltInGadgetManager {
   rpc RunBuiltInGadget(stream BuiltInGadgetControlRequest) returns (stream GadgetEvent) {}
 }
 
+message InstallGadgetInstanceRequest {
+  GadgetInstance gadgetInstance = 1;
+  int32 eventBufferLength = 2;
+}
+
+message InstallGadgetInstanceResponse {
+  int32 result = 1;
+  GadgetInstance gadgetInstance = 2;
+}
+
+message ListGadgetInstancesRequest {}
+
+message GadgetInstance {
+  string id = 1;
```

Comment:
We haven't fully agreed on that, yet (see for example [here](https://github.com/inspektor-gadget/inspektor-gadget/pull/2960#discussion_r1739130252)).

Reply:
If we want `GadgetInstance` to be configuration and not state, it cannot be chosen by the server because it might come from a Helm chart or other user-written template. Then, the server just has to reject the instantiation of a gadget with an id that is already used.

Reply:
I've changed it so it's now done by the client (but will still be filled out by the server if the client sends an empty ID) - that is also a requirement for the `ig --daemon` mode, as there's no centralized store that is synchronized, so each server would just apply it's own ID and you wouldn't be able to find it again.

I opted for overwriting existing instances instead of failing, but I'm not totally sure about it. It's probably better to fail or at least warn.

Reply:
> I opted for overwriting existing instances instead of failing, but I'm not totally sure about it. It's probably better to fail or at least warn.

I've made it fail now if there's already an instance with the given id.